### DISCUSSION TITLE
Import Action with non existant relationship resource

### DISCUSSION THREAD

**Discussion by sambeauprak** [filamentphp/filament#13193] [Help]
### Package

Actions

### Package Version

v3.0.0

### How can we help you?

Hello there,

I need to create an importer through the import action flow. I have a Subscription resource which has a Customer relation.
If I set the Customer ID into the CSV file, it works.

But now, I need to provides the Customer Resource field into the CSV and create the Customer before the Subscription.

How could I do that with the importer?

Thanks by advance

**Comment by pavelpichardo** [comment]
You can use the beforeCreate method inside the importer class:

```
.... getColumns function
----

protected function beforeCreate(Model $record, array $data): Model
  {
        // First, create or find the customer based on email or whatever
        $customer = Customer::firstOrCreate(
            ['email' => $data['customer_email']],
            [
                'name' => $data['customer_name'],
            ]
        );
        // Associate the customer ID with the subscription
        $record->customer_id = $customer->id;
        return $record;
    }
  ```
